tensorflow_probability/python/distributions/continuous_bernoulli.py

```diff
@@ -0,0 +1,454 @@
+        )
+        return unbounded_cdfs * indicators0 + indicators1
+
+    def _log_survival_function(self, x):
+        return tf.math.log(self._survival_function(x))
+
+    def _survival_function(self, x):
+        return 1.0 - self._cdf(x)
+
+    def _outcome_log_lams(self):
+        if self._logits is None:
+            lam = tf.convert_to_tensor(self._lams)
+            return tf.math.log1p(-lam), tf.math.log(lam)
+        s = tf.convert_to_tensor(self._logits)
+        # softplus(s) = -Log[1 - p]
+        # -softplus(-s) = Log[p]
+        # softplus(+inf) = +inf, softplus(-inf) = 0, so...
+        #  logits = -inf ==> log_probs0 = 0, log_probs1 = -inf (as desired)
+        #  logits = +inf ==> log_probs0 = -inf, log_probs1 = 0 (as desired)
+        return -tf.math.softplus(s), -tf.math.softplus(-s)
+
+    def _entropy(self):
+        log_lams0, log_lams1 = self._outcome_log_lams()
+        return (
+            self._mean() * (log_lams0 - log_lams1)
+            - self._cont_bern_log_norm()
+            - log_lams0
+        )
+
+    def _mean(self):
+        lams = self._lams_parameter_no_checks()
+        cut_lams = tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            lams,
+            self._lims[0] * tf.ones_like(lams),
+        )
+        mus = cut_lams / (2.0 * cut_lams - 1.0) + 1.0 / (
+            2.0 * tf.math.atanh(1.0 - 2.0 * cut_lams)
+        )
+        taylor = (
+            0.5 + (lams - 0.5) / 3.0 + 16.0 / 45.0 * tf.math.pow(lams - 0.5, 3)
+        )
+        return tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            mus,
+            taylor,
+        )
+
+    def _variance(self):
+        lams = self._lams_parameter_no_checks()
+        cut_lams = tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            lams,
+            self._lims[0] * tf.ones_like(lams),
+        )
+        vars = cut_lams * (cut_lams - 1.0) / tf.math.pow(
+            1.0 - 2.0 * cut_lams, 2
+        ) + 1.0 / tf.math.pow(2.0 * tf.math.atanh(1.0 - 2.0 * cut_lams), 2)
+        taylor = (
+            1.0 / 12.0
+            - tf.math.pow(lams - 0.5, 2) / 15.0
+            - 128.0 / 945.0 * tf.math.pow(lams - 0.5, 4)
+        )
+        return tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            vars,
+            taylor,
+        )
+
+    def _quantile(self, p):
+        lams = self._lams_parameter_no_checks()
+        cut_lams = tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            lams,
+            self._lims[0] * tf.ones_like(lams),
+        )
+        return tf.where(
```

Comment:
You can make these conditionals simpler by using overrides:

`tf.where(lams < self._lims[0] | lams > self._lims[1], FOO, BAR)`